packages/core/src/domain/session/sessionStore.ts

```diff
@@ -31,47 +32,65 @@ export function startSessionStore<TrackingType extends string>(
   computeSessionState: (rawTrackingType?: string) => { trackingType: TrackingType; isTracked: boolean }
 ): SessionStore {
   const renewObservable = new Observable<void>()
-  const sessionCookie = cacheCookieAccess(SESSION_COOKIE_NAME, options)
-  let inMemorySession = retrieveActiveSession(sessionCookie)
+  let sessionCache: SessionState = retrieveActiveSession(options)
 
   const { throttled: expandOrRenewSession } = utils.throttle(
     monitor(() => {
-      sessionCookie.clearCache()
-      const cookieSession = retrieveActiveSession(sessionCookie)
+      const cookieSession = retrieveActiveSession(options)
       const { trackingType, isTracked } = computeSessionState(cookieSession[productKey])
       cookieSession[productKey] = trackingType
       if (isTracked && !cookieSession.id) {
         cookieSession.id = utils.generateUUID()
         cookieSession.created = String(Date.now())
       }
       // save changes and expand session duration
-      persistSession(cookieSession, sessionCookie)
+      persistSession(cookieSession, options)
 
-      // If the session id has changed, notify that the session has been renewed
-      if (isTracked && inMemorySession.id !== cookieSession.id) {
-        inMemorySession = { ...cookieSession }
+      if (isTracked && sessionCacheOutdated(cookieSession)) {
+        sessionCache = { ...cookieSession }
         renewObservable.notify()
       }
-      inMemorySession = { ...cookieSession }
+      sessionCache = { ...cookieSession }
     }),
     COOKIE_ACCESS_DELAY
   )
 
+  const cookieWatch = setInterval(() => {
+    const cookieSession = retrieveActiveSession(options)
+    if (sessionCacheOutdated(cookieSession)) {
+      clearSessionCache()
+    }
+  }, COOKIE_ACCESS_DELAY)
+
   function expandSession() {
-    sessionCookie.clearCache()
-    const session = retrieveActiveSession(sessionCookie)
-    persistSession(session, sessionCookie)
+    const cookieSession = retrieveActiveSession(options)
+    if (sessionCacheOutdated(cookieSession)) {
+      clearSessionCache()
+    } else {
+      persistSession(cookieSession, options)
+    }
+  }
+
+  function sessionCacheOutdated(cookieSession: SessionState) {
+    return sessionCache.id !== cookieSession.id
+  }
+
+  function clearSessionCache() {
+    sessionCache = {}
   }
 
   function retrieveSession() {
```

Comment:
I would name it `retrieveSessionCache` to be unified with other namings and to disambiguate with the other `retrieveSession` function below

Reply:
Not in favor of exposing the fact that we use a cache to consumers.
I am going with `retrieveSession` -> `getSession`

Reply:
Sure, me neither. What about renaming the function internally to harmonize all cache-related functions namings, but still return it as  `retrieveSession`?

Reply:
I have removed the internal function in f353f77.
let me know what you think